app/src/main/java/com/weatherxm/service/MessagingService.kt

```diff
@@ -51,7 +52,11 @@ class MessagingService : FirebaseMessagingService() {
      * FCM registration token is initially generated so this is where you would retrieve the token.
      */
     override fun onNewToken(token: String) {
-        Timber.d("Refreshed Firebase token: $token")
+        Timber.d("Refreshed FCM token: $token")
+        /**
+         * Init and invoke the work manager to update FCM token in the server
+         */
+        RefreshFcmApiWorker.initAndRefreshToken(applicationContext, token)
```

Comment:
Maybe I'm wrong, but are we sure this object will not be deallocated from the memory before finishes thje job?
I'm saying this because no one holds a ref to this object

Reply:
No, [Workers](https://developer.android.com/reference/androidx/work/Worker) are the proper way of doing work in the background in Android without having to rely on anything else, and we use `applicationContext` here as well.

Taken from the [documentation](https://developer.android.com/reference/androidx/work/Worker):

> A class that performs work synchronously on a background thread provided by [WorkManager](https://developer.android.com/reference/androidx/work/WorkManager).
> 
> If you need to do your work asynchronously or call asynchronous APIs, you should use [ListenableWorker](https://developer.android.com/reference/androidx/work/ListenableWorker).

Also another documentation here regarding [WorkManager](https://developer.android.com/reference/androidx/work/WorkManager):

> WorkManager is the recommended library for persistent work. Scheduled work is guaranteed to execute sometime after its [Constraints](https://developer.android.com/reference/androidx/work/Constraints) are met.
> 
> All work must be done in a [ListenableWorker](https://developer.android.com/reference/androidx/work/ListenableWorker) class. A simple implementation, [Worker](https://developer.android.com/reference/androidx/work/Worker), is recommended as the starting point for most developers. With the optional dependencies, you can also use CoroutineWorker or RxWorker. All background work is given a maximum of ten minutes to finish its execution. After this time has expired, the worker will be signalled to stop.

So it's safe to assume that this code will run no matter what :)

Reply:
I see, WorkManager holds a ref on the task.

Reply:
I think we are good to go

Reply:
Yup, ok. Let's wait for the backend to implement the API endpoint and review it again then?